tmt/beakerlib.py

```diff
@@ -236,7 +272,11 @@ def fetch(self):
                     f"for a deep library ({error}).")
 
 
-def dependencies(original_require, original_recommend=None, parent=None):
+def dependencies(
+    original_require: List[str],
+    original_recommend: Optional[List[str]] = None,
+    parent: Optional[CommonWithLibraryCache] = None
```

Comment:
I think this will lead to `CommonWithLibraryCache` leaking outside of this module. Whoever calls `dependencies()` function would have to apply `cast(CommonWithLibraryCache, ...)` to arguments passed to `dependencies()`.

I'd recommend keeping `CommonWithLibraryCache` fully inside the new `_library_cache` property, while `parent` would remain `Common`, compatible with the rest of the codebase. That way, the fact there is a special attribute involved would be hidden from anything outside of `Library._library_cache` property.

Reply:
ok